### DISCUSSION TITLE
how to manage boolean searches on tables

### DISCUSSION THREAD

**Discussion by marco-santulli** [Budibase/budibase#4896] [General]
Hi,

we have a few situations where we need many filters on tables, including booleans.  Imagine you want to look at a table containing records and you want to see only the "expired" records where "expired" is a boolean on the record.
Typically you want search criteria to be active only when you "activate" them.  Ideally one could use a dropdown with "All", "Expired", "Valid" or something like that..

The issue here is that if you put that in the "filtering" criteria for the boolean, then i cannot show "all records" any more.

a workaround could be to use the new formula column, but I think it doesn't work with filtering yet (or maybe I am doing something wrong) it would be good if we found a better way to do it? does anybody have suggestions?

**Comment by mjashanks** [comment]
Hey Marco.

You should be able to use a formula. It needs to be a "static" formula - meaning that the value is set everytime the record is saved - i.e. on the way in. because it is a saved value, you can run filters on it. 

The drawback of a static formula is that it does not work for relative date calculations. For example, if an "Invoice" is saved with a "Due Date" - you cannot create a formula that will tell you when the invoice is overdue.


Another potential option for you is to use conditions that change your "Define Filters" depending on certain state of your filter fields.

**Reply by marco-santulli** [reply]
Hi Mike,

thank you, I think the static formula works but it pollutes the DB.

I like the conditions option very much, I didn't notice it before, I think it will do the job nicely thank you.  
I have seen on another tool, appsmith, they have created a "show and tell" area, where one can document these solutions, I think it would be useful for budi too, because these nuggets often open people's capability to solve things autnomously, what do you think?

M

**Reply by mjashanks** [reply]
I agree, we do have a show and tell already, but there's not a whole lot in there - https://github.com/Budibase/budibase/discussions/categories/show-and-tell

totally agree though, we need to start building up a list of recipes

**Reply by NicolasFabre81** [reply]
I'm facing the same issue with my postgreSQL table, I think it has to be corrected to be able to make filters on boolean fields.
For the moment the workaround would be to use string fields instead of booleans.
